main.tf

```diff
@@ -1,3 +1,17 @@
-resource "random_pet" "template" {
-  length = var.length
+locals {
+  users = { for u in var.users : u.username => u }
+  roles = {
+    "standard"  = data.datadog_role.standard.id
```

Comment:
👍  Looks good, but we don't have the data yet. I think you're in progress and I'm looking at this too soon 😅